### PR TITLE
HSEARCH-4094 JSR-352 documentation should mention that an additional dependency is necessary

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-4094

### CODE REVIEW

File: documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc [hunk 1]
```diff
@@ -179,7 +179,7 @@ You will need to add this dependency:
 [source, XML, subs="+attributes"]
 ----
 <dependency>
-   <groupId>org.hibernate</groupId>
+   <groupId>org.hibernate.search</groupId>
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc:182]: Nice catch.

File: documentation/src/main/asciidoc/reference/mapper-orm-indexing-jsr352.asciidoc [hunk 2]
```diff
@@ -399,8 +423,7 @@ an entity manager factory by its persistence unit name when using the mass index
 [[jsr-352-emf-other-implementation]]
 === Other DI-enabled JSR-352 implementations
 
-If you want to use a different JSR-352 implementation that happens to allow dependency injection,
-you can use `hibernate-search-mapper-orm-batch-jsr352-core` under the following conditions:
+If you want to use a different JSR-352 implementation:
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/reference/mapper-orm-indexing-jsr352.asciidoc:426]:
```suggestion
If you want to use a different JSR-352 implementation that happens to allow dependency injection:
```

This is really only necessary if the batch runtime supports DI.

File: documentation/src/main/asciidoc/reference/mapper-orm-indexing-jsr352.asciidoc [hunk 3]
```diff
@@ -30,6 +54,8 @@ include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/indexing/Hibe
 
 == Job Parameters
 
+The following table will contain all the job parameters we can use to customize the reindexing batch processes.
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/reference/mapper-orm-indexing-jsr352.asciidoc:57]:
```suggestion
The following table contains all the job parameters you can use to customize the mass-indexing job.
```

Not sure this is really necessary, but as you wish.

**Review by yrodiere — CHANGES_REQUESTED** [review]
Thanks! Only a few comments. Feel free to merge after addressing them.

File: documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc (hunk 1 above)
Comment (fax4ever) [reply documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc:182]: thank you

**Comment by fax4ever** [comment]
Applied the suggestions and merged. Thanks!

**Pull request merged**